src/main/java/zowe/client/sdk/zosfiles/uss/methods/UssExtAttr.java

```diff
@@ -9,5 +9,87 @@
  */
 package zowe.client.sdk.zosfiles.uss.methods;
 
+import kong.unirest.json.JSONObject;
+import zowe.client.sdk.core.ZosConnection;
+import zowe.client.sdk.rest.PutJsonZosmfRequest;
+import zowe.client.sdk.rest.Response;
+import zowe.client.sdk.rest.ZosmfRequest;
+import zowe.client.sdk.rest.ZosmfRequestFactory;
+import zowe.client.sdk.rest.type.ZosmfRequestType;
+import zowe.client.sdk.utility.FileUtils;
+import zowe.client.sdk.utility.ValidateUtils;
+import zowe.client.sdk.zosfiles.ZosFilesConstants;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Provides Unix System Services (USS) extattr functionality
+ *
+ * @author James Kostrewski
+ * @version 2.0
+ */
 public class UssExtAttr {
+
+    private final ZosConnection connection;
+    private ZosmfRequest request;
+
+    /**
+     * UssCopy Constructor
+     *
+     * @param connection connection information, see ZosConnection object
+     * @throws Exception processing error
+     */
+    public UssExtAttr(ZosConnection connection) {
+        ValidateUtils.checkConnection(connection);
+        this.connection = connection;
+    }
+
+    /**
+     * Alternative UssCopy constructor with ZoweRequest object. This is mainly used for internal code
+     * unit testing with mockito, and it is not recommended to be used by the larger community.
+     *
+     * @param connection connection information, see ZosConnection object
+     * @param request    any compatible ZoweRequest Interface object
+     * @throws Exception processing error
+     */
+    public UssExtAttr(ZosConnection connection, ZosmfRequest request) throws Exception {
+        ValidateUtils.checkConnection(connection);
+        ValidateUtils.checkNullParameter(request == null, "request is null");
+        this.connection = connection;
+        if (!(request instanceof PutJsonZosmfRequest)) {
+            throw new Exception("PUT_JSON request type required");
+        }
+        this.request = request;
+    }
+
+    /**
+     * Extends the attributes of a file or directory
+     *
+     * @param set   one of more of the following: alps
+     * @param reset one or more of the following: alps
+     */
+    public Response extendAttributes(String targetPath, String set, String reset) throws Exception {
+        ValidateUtils.checkNullParameter(targetPath == null, "targetPath is null");
+        ValidateUtils.checkIllegalParameter(targetPath.trim().isEmpty(), "targetPath not specified");
+
+        final String url = "https://" + connection.getHost() + ":" + connection.getZosmfPort()
+                + ZosFilesConstants.RESOURCE + ZosFilesConstants.RES_USS_FILES + FileUtils.validatePath(targetPath);
+
+        final Map<String, String> jsonMap = new HashMap<>();
+        jsonMap.put("request", "extattr");
+        if (set != null && !set.trim().isEmpty())
+            jsonMap.put("set", set);
+        if (reset != null && !reset.trim().isEmpty())
+            jsonMap.put("reset", reset);
```

Comment:
this doesnt look pretty the if checks.. I suggest rework this method to execute either a set or reset action, require one to be specified. and create a new method call display(String targetPath) to execute the request without a set and reset specified which would just return a string output of the display attributes as noted in the IBM web documentation.